x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -105,102 +60,190 @@ func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 func RandMsgWithdrawPosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgWithdrawPosition, error) {
 	rand := sim.GetRand()
 	// get random pool
-	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	// positionCreator creates the position with pool denoms
 	// get random user address with the pool denoms
 	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
 	if !senderExists {
 		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
+	// get a random Position
+	positions, err := k.GetUserPositions(ctx, sender.Address, clPool.GetId())
 	if err != nil {
 		return nil, fmt.Errorf("position does not exist")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	if len(positions) < 1 {
+		return nil, fmt.Errorf("user doesnot have any positions")
 	}
```

Comment:
The flow should be the following:

1. Get a random CL pool
2. Utilize the PoolId to PositionId mapping. This will allow you to iterate that store and select a random position ID form that list
3. Use this position to attempt to withdraw

What you implemented does the following:
1. Gets a random address
2. Hopes the address has a position

^ this will result in lots of failed attempts to withdraw, while the first option will fail much less

Reply:
i see, the poolId -> positionid mapping flow was def very clean! 

changed!